### PR TITLE
Vultr: Introducing vr_sshkey_facts module

### PR DESCRIPTION
**Pull request by Spredzy**
##### SUMMARY

This commit introduces a new module called vr_sshkey_facts.

This module aims to return the list of SSH keys avaiable in Vultr.

Sample available here:

```
"vultr_sshkey_facts": {
  "me@home": {
    "date_created": "2018-07-10 14:49:13",
    "id": "5b43c760d7d84",    
    "name": "me@home",
    "ssh_key": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC+ZFQv3MyjtL1BMpSA0o0gIkzLVVC711rthT29hBNeORdNowQ7FSvVWUdAbTq00U7Xzak1ANIYLJyn+0r7olsdG4XEiUR0dqgC99kbT/QhY5mLe5lpl7JUjW9ctn00hNmt+TswpatCKWPNwdeAJT2ERynZaqPobENgewrwerqewqIVew7qFeZygxsPVn36EUr2Cdq7Nb7U0XFXh3x1p0v0+MbL4tiJwPlMAGvFTKIMt+EaA+AsRIxiOo9CMk5ZuOl9pT8h5vNuEOcvS0qx4v44EAD2VOsCVCcrPNMcpuSzZP8dRTGU9wRREAWXngD0Zq9YJMH38VTxHiskoBw1NnPz me@home"
  }
}
```

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

  - vr_sshkey_facts

##### ANSIBLE VERSION

 - devel

##### ADDITIONAL INFORMATION
  
- N/A

### CODE REVIEW

**Review by gundalow — APPROVED** [review]

File: lib/ansible/modules/cloud/vultr/vr_sshkey_facts.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      type: string
+      sample: "https://api.vultr.com"
+ansible_facts:
+  description: Response from Vultr API
+  returned: success
+  type: complex
+  contains:
+    "vultr_sshkey_facts": {
+      "me@home": {
+        "SSHKEYID": "5abf426403479",
```
Comment (resmo) [inline lib/ansible/modules/cloud/vultr/vr_sshkey_facts.py]: I would like to suggest to streamline the keys with the current modules. vultr is not very consistent in the keys and types of returns. Sometimes an id is a string, sometimes a int. etc.

The return should look like:

~~~json
"vultr_sshkey_facts": {
  "id": "5abf426403479",
  "date_created": "2018-02-24 15:04:01",
  "name": "me@home",
  "ssh_key": "ssh-rsa AAAAB3Nz...NnPz me@home"
}
~~~

**Comment by Spredzy** [comment]
@resmo commit update to reflect last comment.

**Comment by Spredzy** [comment]
@resmo: Redefined using commit https://github.com/ansible/ansible/pull/42615/commits/9b428394dd150bb2bb0c4200a72b96f5e29ecdd6 and added tests as discussed on IRC

File: lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      type: string
+      sample: "https://api.vultr.com"
+ansible_facts:
+  description: Response from Vultr API
+  returned: success
+  type: complex
+  contains:
+    "vultr_ssh_key_facts": [
+      {
+        "SSHKEYID": "5abf426403479",
```
Comment (resmo) [inline lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py]: the key is now `id`

~~~
        "id": "5abf426403479",
~~~

File: lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+)
+
+
+class AnsibleVultrSSHKeyFacts(Vultr):
+
+    def __init__(self, module):
+        super(AnsibleVultrSSHKeyFacts, self).__init__(module, "vultr_ssh_key_facts")
+
+        self.returns = {
+            'SSHKEYID': dict(key='id', convert_to='int'),
```
Comment (resmo) [inline lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py]: remove `convert_to='int'`as it does not look like it can be converted.

**Review by resmo — CHANGES_REQUESTED** [review]
some minors, otherwise good to me

File: lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py (hunk 3 above)
Comment (Spredzy) [reply lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py]: Sorry wrong copy/paste from other vr_*_facts module. Indeed, not making sense here. Will fix.

File: lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py (hunk 2 above)
Comment (Spredzy) [reply lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py]: Its actually both. "id" and "SSHKEYID", right?

Comment (resmo) [reply lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py]: I see, I missed this in 577427b1c20dd27bdcd42e417d9ffccfb669b374 that the origin key is also going to be returned as we do not delete the key after "renaming" it. This is going to be done with all the current modules, so every module would have a new return

My intention was, to just rename the key to something "simple": *id*.
Let's delete the previous key in resource in module_utils/vultr.py once renamed, ok?

~~~patch
diff --git a/lib/ansible/module_utils/vultr.py b/lib/ansible/module_utils/vultr.py
index 0dd185c73c..f52052502c 100644
--- a/lib/ansible/module_utils/vultr.py
+++ b/lib/ansible/module_utils/vultr.py
@@ -238,6 +238,7 @@ class Vultr:
 
                 if 'key' in config:
                     resource[config['key']] = resource[search_key]
+                    del resource[search_key]
 
         return resource
~~~

Comment (Spredzy) [reply lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py]: done

Comment (resmo) [reply lib/ansible/modules/cloud/vultr/vr_ssh_key_facts.py]: docs, hint, hint.

~~~diff
-        "SSHKEYID": "5abf426403479",
~~~

**Review by resmo — APPROVED** [review]
shipit

**Pull request merged**